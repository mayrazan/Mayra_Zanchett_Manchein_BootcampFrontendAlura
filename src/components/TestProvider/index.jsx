import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';

const TestProvider = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

TestProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestProvider;