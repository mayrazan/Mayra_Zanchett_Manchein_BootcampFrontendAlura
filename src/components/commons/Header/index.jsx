import React from 'react';
import styled, { css } from 'styled-components';
import breakpointsMedia from '../../../theme/utils/breakpointsMedia';
import Logo from '../Logo';
import NavBar from '../NavBar';

const SectionContainer = styled.section`
  min-height: 40px;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.background.navMain.color};
  padding: 0 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  ${breakpointsMedia({
    md: css`
      justify-content: space-around;
    `,
  })}
  border-top: 2px solid ${({ theme }) => theme.colors.background.nav.color};
`;

const Header = () => (
  <SectionContainer>
    <Logo />
    <NavBar />
  </SectionContainer>
);

export default Header;
