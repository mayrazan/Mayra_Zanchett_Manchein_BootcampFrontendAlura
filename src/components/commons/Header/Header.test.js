import React from 'react';
import { useRouter } from 'next/router';
import { render, screen } from '../../TestProvider/testUtils';
import Header from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('<Header />', () => {
  test('renders header', () => {
    useRouter.mockImplementation(() => ({ pathname: '/' }));

    render(<Header />);
    const textHeader = screen.getByText(/<home \/>/i);
    expect(textHeader).toBeInTheDocument();
    expect(textHeader).toHaveClass('active');
    expect(textHeader).toMatchSnapshot();
  });
});
