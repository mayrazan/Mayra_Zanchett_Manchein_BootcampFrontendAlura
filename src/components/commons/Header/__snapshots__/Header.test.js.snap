// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders header 1`] = `
.c0 {
  color: inherit;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
}

.c0:hover,
.c0:focus {
  opacity: 0.5;
}

.c1 {
  color: #ffffff;
}

@media screen and (min-width:0px) {

}

@media screen and (min-width:768px) {
  .c1 {
    font-size: 38px;
    font-weight: normal;
    line-height: 46px;
  }
}

@media screen and (min-width:0px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:768px) {

}

<a
  class="c0 c1 active"
  color="contrastText"
  href="/"
  rel="noreferrer"
>
  &lt;Home /&gt;
</a>
`;
